### ISSUE TITLE
task-based LR scheduling, finer-grained val reporting

### ISSUE THREAD

**Issue by TimotheeMickus** [Helsinki-NLP/mammoth#33]
- [ ] break down accuracy and validation loss per task when reporting stats
- [ ] introduce validation-based LR scheduling ([reduce LR on plateau](https://pytorch.org/docs/stable/generated/torch.optim.lr_scheduler.ReduceLROnPlateau.html)) that's properly adjusted depending on the performances of individual modules
